### PR TITLE
Change DeferredActionsQueue to a Min-Heap

### PR DESCRIPTION
**Pull request by Lynesth**
A few things to note:
It doesn't change behavior at the moment but will permit better granularity in the future.
The Priority enum will change in an upcoming PR. Right now it's just like this to keep the cards behavior.

It's easier to read and understand this PR commit by commit.

### CODE REVIEW

**Review by kberg — APPROVED** [review]

**Review by kberg — COMMENTED** [review]
Wait, wrong, wait.

File: src/cards/venusNext/SponsoredAcademies.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export class SponsoredAcademies extends Card {
   }
 
   public play(player: Player, game: Game) {
-    game.defer(new DiscardCards(player));
+    game.defer(new DiscardCards(player), -1);
```
Comment (kberg) [inline src/cards/venusNext/SponsoredAcademies.ts:35]: Please give a value to this priority in the enum.

**Review by kberg — APPROVED** [review]
Having reviewed in commit order, you're right, it's pretty clear. One change requested. Nice work!

@bafolts may have thoughts about the new node package, but I'd like to use it and abandon my hand-built multiset.

File: src/cards/venusNext/SponsoredAcademies.ts (hunk 1 above)
Comment (Lynesth) [reply src/cards/venusNext/SponsoredAcademies.ts]: It's done.
Names in the enum are not final at all, it was just to keep the current behavior, they will be reworked in an upcoming PR.

**Comment by bafolts** [comment]
I am cautious of taking on new dependencies. As the package only has one dependent and seems it is written with good goals in mind it seems fine. This will not impact the client side build size as it only runs on the server. I would replace the multiset we have once taking on this dependency. If we only want to use the heap there is also a stand alone priority queue with 0 dependents and more weekly downloads.

https://www.npmjs.com/package/tinyqueue

**Review by bafolts — APPROVED** [review]

**Comment by Lynesth** [comment]
I didn't go with TinyQueue as it seemed much less performant based on a benchmark I unfortunatly cannot find anymore.

**Comment by kberg** [comment]
> I am cautious of taking on new dependencies. As the package only has one dependent and seems it is written with good goals in mind it seems fine. This will not impact the client side build size as it only runs on the server. I would replace the multiset we have once taking on this dependency. If we only want to use the heap there is also a stand alone priority queue with 0 dependents and more weekly downloads.
> 
> https://www.npmjs.com/package/tinyqueue

Don't be so certain. Deferred actions code absolutely appears in the browser. That's why I was saying we should focus sometime on extracting what code is common between client and server.

**Pull request merged**